src/dfmt.d

```diff
@@ -197,6 +197,28 @@ private:
                     newline();
                     break;
                 }
+                else if (current.type == tok!",")
+                {
+                    // compute length until next , or ;
+                    int length_of_next_chunk = INVALID_TOKEN_LENGTH;
+                    for (size_t i=index+1; i<tokens.length; i++)
+                    {
+                        if (tokens[i].type == tok!"," || tokens[i].type == tok!";")
+                            break;
+                        const len = tokenLength(i);
+                        assert (len >= 0);
+                        length_of_next_chunk += len;
+                    }
```

Comment:
This whole approach could be generalized into some clever algorithm which *rather* breaks according to precedences.  Also see issue #7